cognite/client/_api/datapoints.py

```diff
@@ -604,6 +623,8 @@ def retrieve(
         end: int | str | datetime | None = None,
         aggregates: Aggregate | str | list[Aggregate | str] | None = None,
         granularity: str | None = None,
+        target_unit: str | None = None,
+        target_unit_system: str | None = None,
```

Comment:
To mimic the API, these should actually be passed as part of individual settings, not at the top level. I still agree with adding here, as the default for any non-specified. But! Passing as part of the individual settings doesn't work:
```py
>>> client.time_series.data.retrieve(
>>>     id={"id": created_timeseries.id, "target_unit": "temperature:k"})
KeyError: "Dict provided by argument `id` included key(s) not understood: ['target_unit']. Required key: `id`. Optional: ['start', 'end', 'granularity', 'aggregates', 'include_outside_points', 'ignore_unknown_ids', 'limit']."

```

Reply:
(Let me know if you want me to do this part)

Reply:
Pushed a fix in 9b6accd.